intelmq/bots/outputs/cif3/output.py

```diff
@@ -0,0 +1,233 @@
+  - fireball: int, used to batch events before submitting to a CIFv3 instance
+        (default is 500 per batch, use 0 to disable batch and send each event as received)
+  - http_verify_cert: bool, used to tell whether the CIFv3 instance cert should be verified
+        (default true, but can be set to false if using a local test instance)
+
+Example (of some parameters in JSON)::
+
+    "add_feed_provider_as_tag": true,
+    "cif3_additional_tags": ["intelmq"]
+
+
+"""
+import ujson as json
+from datetime import datetime
+
+from intelmq.lib.bot import OutputBot
+from intelmq.lib.exceptions import IntelMQException, MissingDependencyError
+
+try:
+    from cifsdk.client.http import HTTP as HttpClient
+    from csirtg_indicator import Indicator
+except ImportError:
+    HttpClient = None
+    import_fail_reason = 'import'
+except SyntaxError:
+    HttpClient = None
+    import_fail_reason = 'syntax'
+
+INTELMQ_TO_CIF_FIELDS_MAP = {
+    'source.ip': 'indicator',
+    'source.fqdn': 'indicator',
+    'source.url': 'indicator',
+    'source.network': 'indicator',
+    'source.port': 'port',
+    'feed.url': 'reference',
+    'time.source': 'lasttime',
+    'time.observation': 'reporttime',
+    'event_description.text': 'description',
+    'event_description.url': 'reference',
+    'malware.hash.md5': 'indicator',
+    'malware.hash.sha1': 'indicator',
+    'malware.hash.sha256': 'indicator',
+    'malware.name': 'description',
+    'protocol.application': 'application',
+    'protocol.transport': 'protocol',
+    'tlp': 'tlp',
+}
+
+INTELMQ_CLASSIFICATION_TO_CIF_TAGS_MAP = {
+    'c2-server': 'botnet,c2',
+    'undetermined': 'suspicious',
+    'brute-force': 'bruteforce',
+    'other': 'suspicious',
+}
+
+
+class CIF3OutputBot(OutputBot):
+    """
+    Submits indicators to a CIFv3 instance
+
+
+    IntelMQ-Bot-Name: CIFv3 API
+    """
+    add_feed_provider_as_tag: bool = False
+    cif3_feed_confidence: float = 5
+    cif3_static_confidence: bool = False
+    cif3_additional_tags = []
+    cif3_token: str = None
+    cif3_url: str = None
+    fireball: int = 500
+    http_verify_cert: bool = True
+
+    _is_multithreadable = False
+
+    def init(self):
+        if HttpClient is None and import_fail_reason == 'syntax':
+            raise MissingDependencyError(
+                "cifsdk",
+                version='3.0.0rc4,<4.0'
+            )
+        elif HttpClient is None:
+            raise MissingDependencyError('cifsdk', version='3.0.0rc4,<4.0')
+
+        self.cif3_url = self.cif3_url.rstrip('/')
+
+        self.logger.info(f"Connecting to CIFv3 instance at {self.cif3_url!r}.")
+        self.cli = HttpClient(self.cif3_url,
+                                self.cif3_token,
+                                verify_ssl=self.http_verify_cert)
+
+        try:
+            _ = self.cli.ping(write=True)
+        except Exception as err:
+            raise IntelMQException(f"Error connecting to CIFv3 instance: {err}")
```

Comment:
```suggestion
            raise ValueError(f"Error connecting to CIFv3 instance: {err}")
```
or don't catch and re-raise it at all and just let ping raise